src/ddmd/root/file.d

```diff
@@ -77,12 +77,65 @@ nothrow:
     {
         if (len)
             return false; // already read the file
+
+        import core.stdc.string : strcmp;
+        const(char)* name = this.name.toChars();
+        if (strcmp(name, "__stdin.d") == 0)
+        {
+            /* Read from stdin */
+            enum bufIncrement = 128 * 1024;
+            size_t pos = 0;
+            size_t sz = bufIncrement;
+
+            if (!_ref)
+                .free(buffer);
+
+            buffer = null;
+            L1: for (;;)
+            {
+                buffer = cast(ubyte*).realloc(buffer, sz + 2); // +2 for sentinel
+                if (!buffer)
+                {
+                    printf("\tmalloc error, errno = %d\n", errno);
+                    break L1;
+                }
+
+                // Fill up buffer
+                do
```

Comment:
a while more expressive in this case.
Since the condition is supposed to be true on the first run.